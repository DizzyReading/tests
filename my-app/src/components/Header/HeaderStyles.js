import styled from "styled-components";

export const Container = styled.header`
  display: block;
  width: 100%;
  max-height: 85px;
  margin: 0 auto;
  padding: 0 10px;
  background: ${(props) => props.theme.colors.primary1}; ;
`;

Container.displayName = "Container";

export const Wrap = styled.div`
  display: relative;
  width: 100%;
  height: 100%;
  margin: 0 auto;
  max-width: 980px;
`;
export const Logo = styled.div`
  display: block;
  max-width: 120px;
  left: 10px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  --moz-transform: translateY(-50%);
  transform: translateY(-50%);
`;

export const Image = styled.img`
  max-width: 70%;
  margin-top: 9rem;
`;

export const Title = styled.h1`
  color: ${(props) => props.theme.colors.primary1};
`;
