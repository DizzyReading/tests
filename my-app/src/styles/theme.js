import React from "react";
import { ThemeProvider } from "styled-components";
import PropTypes from "prop-types";
import theme from "../themes/default";
import GlobalStyles from "./globals";

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    <GlobalStyles />
    {children}
  </ThemeProvider>
);

Theme.propTypes = {
  children: PropTypes.element.isRequired,
};

export default Theme;
