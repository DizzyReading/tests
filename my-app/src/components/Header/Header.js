/* eslint-disable no-unused-vars */
import React from "react";
import { Container, Wrap, Logo, Image } from "./HeaderStyles";
import logo from "../../assets/graphics/logo.png";
import PropTypes from "prop-types";
const Header = ({ theme, ...props }) => {
  return (
    <Container theme={theme} data-test="headerComponent">
      <Wrap>
        <Logo>
          <Image data-test="logoImg" src={logo} />
        </Logo>
      </Wrap>
    </Container>
  );
};

Header.propTypes = {
  theme: PropTypes.object,
};

export default Header;
