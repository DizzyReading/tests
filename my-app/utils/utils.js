import React from "react";
import checkPropTypes from "check-prop-types";
import theme from "../src/themes/default";
import TestRenderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import { configureStore } from "redux-mock-store";
import postsReducer from "../src/redux/posts/postsSlice";
import { createStoreHook } from "react-redux";
import { applyMiddleware, createStore } from "redux";
import ReduxThunk from "redux-thunk";

export const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component, expectedProps) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    "props",
    component.name
  );

  return propsErr;
};

export const renderWithTheme = (component) => {
  return TestRenderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
};

const middlewares = [ReduxThunk];

export const testStore = (initialState) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(
    createStore
  );
  return createStoreWithMiddleware(postsReducer, initialState);
};
