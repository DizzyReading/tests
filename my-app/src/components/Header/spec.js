import "jest-styled-components";
import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import Header from "./Header";
import toJson from "enzyme-to-json";
import theme from "../../themes/default";
import { findByTestAtrr, renderWithTheme } from "../../../utils/utils";
import { Container, Image, Logo, Wrap } from "./HeaderStyles";
configure({ adapter: new Adapter() });

const setUp = (props = {}) => {
  const component = shallow(<Header theme={theme} {...props}></Header>);

  return component;
};

describe("Header Component", () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  it("It should render without errors and default theme", () => {
    const container = findByTestAtrr(component, "headerComponent");
    expect(container.length).toBe(1);
    expect(toJson(container)).toMatchSnapshot();
  });

  it("It should render a logo", () => {
    // console.log(component.debug());
    const logo = findByTestAtrr(component, "logoImg");
    expect(logo.length).toBe(1);
    expect(toJson(logo)).toMatchSnapshot();
  });
});

describe("Container Block", () => {
  it("It should render correctly with given css", () => {
    const tree = renderWithTheme(<Container></Container>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
