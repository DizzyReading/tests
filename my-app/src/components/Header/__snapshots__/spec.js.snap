// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Block It should render correctly with given css 1`] = `
.c0 {
  display: block;
  width: 100%;
  max-height: 85px;
  margin: 0 auto;
  padding: 0 10px;
  background: #606060FF;
}

<header
  className="c0"
/>
`;

exports[`Header Component It should render a logo 1`] = `
<styled.img
  data-test="logoImg"
  src="test-file-stub"
/>
`;

exports[`Header Component It should render without errors and default theme 1`] = `
<Container
  data-test="headerComponent"
  theme={
    Object {
      "breakpoints": Object {
        "lg": "screen and (max-width: 1024px)",
        "md": "screen and (max-width: 768px)",
        "sm": "screen and (max-width: 640px)",
        "xl": "screen and (max-width: 1280px)",
      },
      "colors": Object {
        "accent1": "hsl(34.9,98.6%,72.9%)",
        "background1": "#D6ED17FF",
        "background2": "hsl(232.7,27.3%,23.7%)",
        "button": "hsl(205.1,100%,36.1%)",
        "primary1": "#606060FF",
      },
      "fonts": Object {
        "main": "Space Grotesk, sans-serif",
        "title": "Space Grotesk, sans-serif",
      },
    }
  }
>
  <styled.div>
    <styled.div>
      <styled.img
        data-test="logoImg"
        src="test-file-stub"
      />
    </styled.div>
  </styled.div>
</Container>
`;
